src/main/java/com/okta/tools/authentication/BrowserAuthentication.java

```diff
@@ -92,6 +92,9 @@ private String getSamlResponseForAws(Document document) {
     }
 
     private Node getAwsStsSamlForm(Document document) {
+        if (document == null) {
+            return null;
+        }
```

Comment:
I don't understand how this NPE is possible. I'm going to look into how to terminate a JavaFX application cleanly. I suspect using System.exit from a listener violates some invariants.

There is already a null check here:
https://github.com/oktadeveloper/okta-aws-cli-assume-role/blob/master/src/main/java/com/okta/tools/authentication/BrowserAuthentication.java#L69

The NPE you got suggests that `webEngine.getDocument()` returns non-**null** and then returns **null** in the same thread of control with no significant intervening code (a method invocation is the only signicant thing that happens):
https://github.com/oktadeveloper/okta-aws-cli-assume-role/blob/master/src/main/java/com/okta/tools/authentication/BrowserAuthentication.java#L83

Is this NPE consistently reproducible? I suspect it may not be on macOS, but JavaFX implementation is platform-specific so exit modes may vary on GNOME or Windows.

Reply:
My bad it came from a bad rebase